content/tokio/tutorial/async.md

```diff
@@ -557,18 +574,42 @@ impl Task {
         // Poll the future
         let _ = future.as_mut().poll(&mut cx);
     }
+
+    // Creates the Task given a Future and send it to execution
+    // for the very first time.
+    fn spawn<F>(future: F, sender: &channel::Sender<Arc<Task>>)
+    where
+        F: Future<Output = ()> + Send + 'static,
+    {
+        let task = Arc::new(Task {
+            future: Mutex::new(Box::pin(future)),
+            executor: sender.clone(),
+        });
+
+        let _ = sender.send(task);
+    }
+
 }
 ```
 
-Two things are happening here. First, `MiniTokio::run()` is implemented. The
-function runs in a loop receiving scheduled tasks from the channel. As tasks are
-pushed into the channel when they are woken, these tasks are able to make
-progress when executed.
+Multiple things are happening here. First, `MiniTokio::run()` is implemented.
+The function runs in a loop receiving scheduled tasks from the channel.
+As tasksare pushed into the channel when they are woken, these tasks are able
+to make progress when executed.
+
+Then the functions `MiniTokio::new()` and `MiniTokio::spawn()` are adjusted for
+working with the channel instead of the `VecDeque`. `MiniTokio::new()` creates
+the channel and sets it on the new `MiniTokio` before returning it.
+`MiniTokio::spawn()` receives the future and pass it to the `Task::spawn`
+function together with the send half of the channel..
```

Comment:
The sentence structure here is a bit complicated.
```suggestion
Additionally, the `MiniTokio::new` and `MiniTokio::spawn` methods are adjusted
to use a channel rather than a `VecDeque`. When new tasks are spawned, they are
given a clone of the sender-part of the channel, which the task can use to
schedule itself on the runtime.
```